contrib/inventory/ovirt4.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+oVirt dynamic inventory script
+=================================
+
+Generates dynamic inventory file for oVirt.
+
+Script will return following attributes for each virtual machine:
+ - id
+ - name
+ - host
+ - cluster
+ - status
+ - description
+ - fqdn
+ - os_type
+ - template
+ - tags
+ - statistics
+ - devices
+
+When run in --list mode, virtual machines are grouped by the following categories:
+ - cluster
+ - tag
+ - status
+
+ Note: If there is some virtual machine which has has more tags it will be in both tag
+       records.
+
+Examples:
+  # Execute update of system on webserver virtual machine:
+
+    $ ansible -i contrib/inventory/ovirt4.py webserver -m yum -a "name=* state=latest"
+
+  # Get webserver virtual machine information:
+
+    $ contrib/inventory/ovirt4.py --host webserver
+
+Author: Ondra Machacek (@machacekondra)
+"""
+
+import argparse
+import ConfigParser
+import os
+import sys
+
+from collections import defaultdict
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    import ovirtsdk4 as sdk
+    import ovirtsdk4.types as otypes
+except ImportError:
+    print('oVirt inventory script requires ovirt-engine-sdk-python >= 4.0.0')
+    sys.exit(1)
+
+
+def parse_args():
+    """
+    Create command line parser for oVirt dynamic inventory script.
+    """
+    parser = argparse.ArgumentParser(
+        description='Ansible dynamic inventory script for oVirt.',
+    )
+    parser.add_argument(
+        '--list',
+        action='store_true',
+        default=True,
+        help='Get data of all virtual machines (default: True).',
+    )
+    parser.add_argument(
+        '--host',
+        help='Get data of specific virtual machine.',
```

Comment:
I'd rephrase comment to: 'Get data of virtual machines running on specified host'